eng/pipelines/templates/stages/platform-matrix.json

```
@@ -10,14 +13,22 @@
     },
     "JavaTestVersion": [ "1.8", "1.21" ],
     "AZURE_TEST_HTTP_CLIENTS": [ "okhttp", "netty" ],
-    "TestFromSource": false,
-    "RunAggregateReports": false,
-    "TestGoals": "surefire:test failsafe:integration-test failsafe:verify",
-    "TestOptions": ""
+    "TestFromSource": {
+      "NotFromSource": { "TestFromSource": false }
+    },
+    "RunAggregateReports": {
+      "$": { "RunAggregateReports": false }
+    },
+    "TestOptions": {
+      "$": { "TestOptions": "" }
+    },
+    "TestGoals": {
+      "TestsOnly": { "TestGoals": "surefire:test failsafe:integration-test failsafe:verify" }
+    }
```

Since all the values are singular, you can just put them in one grouping to make the config easier to read:

```suggestion
    "Options": {
      "NotFromSource_TestsOnly": { 
        "TestFromSource": false,
        "RunAggregateReports": false,
        "TestOptions": "",
        "TestGoals": "surefire:test failsafe:integration-test failsafe:verify"
      }
    }
```